src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -180,6 +199,61 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  def syncInternalCreateCluster(userEmail: WorkbenchEmail,
+                                serviceAccountInfo: ServiceAccountInfo,
+                                googleProject: GoogleProject,
+                                clusterName: ClusterName,
+                                clusterRequest: ClusterRequest): Future[Cluster] = {
+    // Check if the google project has an active cluster with the same name. If not, we can create it
+    dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.getActiveClusterByName(googleProject, clusterName)
+    } flatMap {
+      case Some(existingCluster) =>
+        throw ClusterAlreadyExistsException(googleProject, clusterName, existingCluster.status)
+      case None =>
+        val augmentedClusterRequest = addClusterDefaultLabels(
+          serviceAccountInfo, googleProject, clusterName, userEmail, clusterRequest)
+        val machineConfig = MachineConfigOps.create(clusterRequest.machineConfig, clusterDefaultsConfig)
+        val autopauseThreshold = calculateAutopauseThreshold(
+          clusterRequest.autopause, clusterRequest.autopauseThreshold)
+        val initialCluster = Cluster.create(
+          clusterRequest, userEmail, clusterName, googleProject,
+          serviceAccountInfo, machineConfig, dataprocConfig.clusterUrlBase, autopauseThreshold)
+
+        val attemptToSaveClusterInDb: Future[Cluster] = dbRef.inTransaction(_.clusterQuery.save(initialCluster))
+
+        // For the success case, register the following callbacks...
+        attemptToSaveClusterInDb foreach { cluster =>
+          // Notify the auth provider and kick off the cluster creation on the Google Dataproc side
+          asyncClusterCreate(userEmail, serviceAccountInfo, googleProject, clusterName, augmentedClusterRequest)
+            .onComplete {
```

Comment:
For debuggability's sake, it'd be nice to add logging for when the async portion begins and ends for a cluster. I'd like be able to trace a full cluster creation lifecycle by looking at logs. For example we could log:
1. the API response committed (we have this)
2. the async portion starting
3. the async portion ending
4. the ClusterMonitorActor starting (we have this)
5. the ClusterMonitorActor finishing (we have this)